add calculated usage, split up into files

jdelbick opened the pull request:
- adjusted the file structure: created a /lib folder and split up code into different files: /memory, /cpu, /metrics

- added a `timestamp` attribute to cpu metrics
- all cpu metrics now return an object with the metric/metrics and a timestamp
- added a function to calculate the cpu usage over time: `calculateUsage`
- added Docker app for manual testing and confirmed measurements are accurate


File                     |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-------------------------|----------|----------|----------|----------|-------------------|
All files                |    98.33 |    93.33 |      100 |    98.33 |                   |
 node-cgroup-metrics     |      100 |      100 |      100 |      100 |                   |
  index.js               |      100 |      100 |      100 |      100 |                   |
 node-cgroup-metrics/lib |    98.21 |    93.33 |      100 |    98.21 |                   |
  cpu.js                 |      100 |      100 |      100 |      100 |                   |
  memory.js              |    93.75 |    94.12 |      100 |    93.75 |                44 |
  metrics.js             |      100 |      100 |      100 |      100 |                   |
  utils.js               |      100 |    91.67 |      100 |      100 |                57 |

alexkli commented on lib/metrics.js:
```diff
@@ -0,0 +1,107 @@
+        calculatedUsage: (usageFirst, usageSecond) => {
+            const totalTime = usageSecond.timestamp - usageFirst.timestamp;
+            const deltaNS = usageSecond.timeSinceContainerNS - usageFirst.timeSinceContainerNS;
+            return deltaNS / totalTime * 100
```
The timestamps are in milliseconds, while the measurement is in nano seconds. I think it has to be adjusted.

Would be great to test this actually works with a little sample app and e.g. compare with `docker stats`.

jdelbick replied:
Sounds like a good idea. Should I use a library to mock a docker container?

alexkli replied:
I was thinking of manual testing first. Because this isn't well documented...

Then your existing unit tests would be fine I guess, just the numbers would need adjustment.

jdelbick commented:
@alexkli I changed the output of the `cpu.stats` function so it made a bit more sense. I updated the documentation to make it easier to understand:
- I moved the timestamp generation into the `readMetric` function and renamed the function to `readandFormatMetric` since it now does more than just read.
Now both cpu metrics functions return object/objects of the form:
```
{
          cpuNanosSinceContainerStart: 120234,
         timestamp: 153686574
 }
``` so they can both be passed to the `calculatedUsage` function.

Also I updated the `calculatedUsage` function so it converts the total time to NS before calculating the usage

alexkli commented on README.md:
```diff
@@ -22,16 +22,18 @@ Calculated values:
 
 Calculated CPU values:
-- `calculatedUsage`: takes two instances of calls to `cpuacct.usage` and returns the calculated usage:
+- `calculatedUsage`: takes two instances of calls to `cpuacct.usage` or `cpuacct.stat` and returns the calculated usage:
```
`... returns the calculated usage in percentage of CPU time`:
